### ISSUE TITLE
🐛 Bug Report: Error while upgrading to 1.24.0 at the beginning of the migration

### ISSUE THREAD

**Issue by toughthomas** [backstage/backstage#25876]
### 📜 Description

I am currently running backstage 1.23.3, and i want to update it to 1.24.0, i have noticed the breaking change concerning the backend and trying to do the migration. I have upgraded all the version in package.json using https://backstage.github.io/upgrade-helper/?from=1.23.3&to=1.24.0

and modified my backend` index.ts` file like so : 

```
import { createBackend } from '@backstage/backend-defaults';

const backend = createBackend();
backend.add(import('@backstage/plugin-app-backend/alpha'));
backend.start();
```

But i have a console error : 

`Failed to instantiate service 'core.httpRouter' for 'app' because the factory function threw an error, Error: Failed to instantiate service 'core.auth' for 'app' because the factory function threw an error, Error: ENOENT: no such file or directory, scandir '/Users/xxxx/kry/code/backstage/node_modules/@backstage/backend-defaults/migrations/auth'`

and in fact this directory doesn't exist. The `backend-defaults` module is version 0.2.18.

and i have this error when i go to http://localhost:3000/ : 
<img width="581" alt="Screenshot 2024-08-01 at 10 45 53" src="https://github.com/user-attachments/assets/76282c21-fbe0-4b97-b4c8-56b0bea7b8eb">

### 👍 Expected behavior

It should work as i have followed every steps in https://backstage.io/docs/backend-system/building-backends/migrating/#overview

### 👎 Actual Behavior with Screenshots

see the description above

### 👟 Reproduction steps

have a 1.23.3 version and try to update it to 1.24.0

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by benjdlambert** [comment]
Hmm, I think that it might be better for you to upgrade to the latest version by using `yarn backstage-cli versions:bump` and go from there. The issue is most likely that by changing the `package.json` values manually, there's no de-duping of dependencies, which could lead to incompatibilities in the framework.

**Comment by zeshanziya** [comment]
Yes, I had the same issue while upgrading from 1.23 to 1.24. So I upgraded to 1.29 and followed all the releases notes for migrations.

Here are a few points for your reference:

- Few plugins moved to community plugins.
- Migrate to the new auth system (remove cookie-auth and s2s auth).
- Migrate all backend plugins to use the new auth system.
- Remove `loggerToWinstonLogger` and use `LoggerService`.
- Remove `getVoidLogger` and use `mockService` for testing.
- Migrate from `PluginDatabaseManager` to `DatabaseService`.
- Migrate from `TaskRunner` to `SchedulerServiceTaskRunner`.
- Migrate from `PluginEndpointDiscovery` to `DiscoveryService`.
- TokenManager to `AuthService` and `HttpAuthService`.
- Remove `getBearerTokenFromAuthorizationHeader` and use `httpAuth.credentials`.
- From `BackstageIdentityResponse` to `PolicyQueryUser` for permission policy.
